src/store/modules/layers.store.js

```diff
@@ -429,6 +430,48 @@ const actions = {
     clearPreviewYear({ commit }) {
         commit('setPreviewYear', null)
     },
+    setLayerErrorKey({ commit, getters }, payload) {
+        const { layerId, errorKey } = payload
+        const currentLayer = getters.getActiveLayerById(layerId)
+        if (!currentLayer) {
+            throw new Error(
+                `Failed to update layer error key "${layerId}", layer not found in active layers`
+            )
+        }
+        const updatedLayer = currentLayer.clone()
+        updatedLayer.errorKey = errorKey
+        updatedLayer.hasError = !!errorKey
+        commit('updateLayer', updatedLayer)
+    },
```

Comment:
Shouldn't that only be applied to KML layers?
If that's not what is intended, I would move the declaration of `errorKey` and `hasError` into `AbstractLayer.class.js` instead of `KMLLayer.class.js`

Reply:
`errorKey` and `hasError` were meant for `ExternalLayer` but because for KML we have only one class for internal and external I had to put the flag again in this class. But this is a good idea to move them into the abstract layer even if for the moment they are only used for external layer (internal layer should never have an error ;-) ).